code/modules/client/verbs/viewport.dm

```diff
@@ -0,0 +1,48 @@
+/// Used in OnResize() - Holds the clients prefs.
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
+/// General handler for viewport updates. Takes an argument for an icon size, if not aligning with your preference, updates your preference.
+/client/verb/SetWindowIconSize(val as num|text)
+	set hidden = 1
+	winset(src, "mapwindow.map", "icon-size=[val]")
+	if(prefs && val != prefs.read_preference(/datum/preference/numeric/icon_size))
+		prefs.write_preference(GLOB.preference_entries[/datum/preference/numeric/icon_size], val)
+		prefs.save_preferences()
+	OnResize()
+/// Verb called by the games skin.dmf to update the server to update view.
+/client/verb/OnResize()
+	set hidden = 1
+	if(!prefs?.read_preference(/datum/preference/toggle/widescreen))
+		return
+
+	var/divisor = text2num(winget(src, "mapwindow.map", "icon-size")) || prefs.read_preference(/datum/preference/numeric/icon_size) || world.icon_size
+	if(!isnull(GLOB.lock_client_view_x) && !isnull(GLOB.lock_client_view_y))
+		last_view_x_dim = GLOB.lock_client_view_x
+		last_view_y_dim = GLOB.lock_client_view_y
+	else
+		var/winsize_string = winget(src, "mapwindow.map", "size")
+		last_view_x_dim = GLOB.lock_client_view_x || clamp(ROUND_UP(text2num(winsize_string) / divisor), 15, (CONFIG_GET(number/max_client_view_x)) || (CONFIG_GET(number/max_client_view_x)))
+		last_view_y_dim = GLOB.lock_client_view_y || clamp(ROUND_UP(text2num(copytext(winsize_string,findtext(winsize_string,"x")+1,0)) / divisor), 15, (CONFIG_GET(number/max_client_view_y)) || (CONFIG_GET(number/max_client_view_y)))
+		if(last_view_x_dim % 2 == 0)
+			last_view_x_dim++
+		if(last_view_y_dim % 2 == 0)
+			last_view_y_dim++
```

Comment:
so the 16 height limit is actually 17? only odd values allowed in that config option?

Reply:
Alright SO this is for tile range sent to the client. If you have an even number, you have no true center - 6x6, 3, or 4, your middle, would not be center. However, 5x5 has 3 as a center, and 15x15 has 7,7 as a center. If you're at 16x16, you do not have a middle point, that middle point is between two tiles, and thus impossible to center your view information.